react/features/base/settings/reducer.js

```diff
@@ -0,0 +1,162 @@
+// @flow
+
+import { APP_WILL_MOUNT } from '../../app';
+import JitsiMeetJS, { browser } from '../lib-jitsi-meet';
+import { ReducerRegistry } from '../redux';
+import { PersistenceRegistry } from '../storage';
+import { randomHexString } from '../util';
+
+import { SETTINGS_UPDATED } from './actionTypes';
+
+const logger = require('jitsi-meet-logger').getLogger(__filename);
+
+/**
+ * The default/initial redux state of the feature {@code base/settings}.
+ *
+ * @type Object
+ */
+const DEFAULT_STATE = {};
+
+const STORE_NAME = 'features/base/settings';
+
+/**
+ * Sets up the persistence of the feature {@code base/settings}.
+ *
+ * Note: There are some values inherited from the old Settings.js that needs
+ * session-based storage but not persisted storage, such as cameraDeviceId
+ * and micDeviceId so we utilize the selective storage functionality of the
+ * persistency engine.
+ */
+PersistenceRegistry.register(STORE_NAME, {
+    audioOutputDeviceId: true,
+    avatarID: true,
+    avatarURL: true,
+    cameraDeviceId: false,
+    displayName: true,
+    email: true,
+    localFlipX: true,
+    micDeviceId: false,
+    serverURL: true,
+    startAudioOnly: true,
+    startWithAudioMuted: true,
+    startWithVideoMuted: true
+});
+
+ReducerRegistry.register(STORE_NAME, (state = DEFAULT_STATE, action) => {
+    switch (action.type) {
+    case APP_WILL_MOUNT:
+        return _initSettings(state);
+
+    case SETTINGS_UPDATED:
+        return {
+            ...state,
+            ...action.settings
+        };
+    }
+
+    return state;
+});
+
+/**
+ * Retrieves the legacy profile values regardless of it's being in pre or
+ * post-flattening format.
+ *
+ * @private
+ * @returns {Object}
+ */
+function _getLegacyProfile() {
+    let persistedProfile
+        = window.localStorage.getItem('features/base/profile');
+
+    if (persistedProfile) {
+        try {
+            persistedProfile = JSON.parse(persistedProfile);
+
+            if (persistedProfile && typeof persistedProfile === 'object') {
+                const preFlattenedProfile = persistedProfile.profile;
+
+                return preFlattenedProfile || persistedProfile;
+            }
+        } catch (e) {
+            logger.warn('Error parsing persisted legacy profile', e);
+        }
+    }
+
+    return {};
+}
+
+/**
+ * Inits the settings object based on what information we have available.
+ * Info taken into consideration:
+ *   - Old Settings.js style data
+ *   - Things that we stored in profile earlier but belong here.
+ *
+ * @private
+ * @param {Object} state - The current state of the feature.
+ * @returns {Object}
+ */
+function _initSettings(state) {
```

Comment:
Please call this something else, it sounds like the entire redux store state, but it's not, AFAICT.

Reply:
You mean the parameter name? Changed it, thanks! Or did you mean something else?